modin/core/execution/dispatching/factories/factories.py

```diff
@@ -651,3 +651,72 @@ def prepare(cls):
 @doc(_doc_factory_class, execution_name="experimental remote HdkOnNative")
 class ExperimentalHdkOnCloudnativeFactory(ExperimentalRemoteFactory):
     wrapped_factory = ExperimentalHdkOnNativeFactory
+
+
+@doc(_doc_factory_class, execution_name="PandasOnUnidist")
+class PandasOnUnidistFactory(BaseFactory):
+    @classmethod
+    @doc(_doc_factory_prepare_method, io_module_name="``PandasOnUnidistIO``")
+    def prepare(cls):
+        from modin.core.execution.unidist.implementations.pandas_on_unidist.io import (
+            PandasOnUnidistIO,
+        )
+
+        cls.io_cls = PandasOnUnidistIO
+
+
+@doc(_doc_factory_class, execution_name="experimental PandasOnUnidist")
+class ExperimentalPandasOnUnidistFactory(
+    ExperimentalBaseFactory, PandasOnUnidistFactory
+):
+    @classmethod
+    @doc(
+        _doc_factory_prepare_method, io_module_name="``ExperimentalPandasOnUnidistIO``"
+    )
+    def prepare(cls):
+        from modin.experimental.core.execution.unidist.implementations.pandas_on_unidist.io import (
+            ExperimentalPandasOnUnidistIO,
+        )
+
+        cls.io_cls = ExperimentalPandasOnUnidistIO
+
+    @classmethod
+    @doc(
+        _doc_io_method_raw_template,
+        source="CSV files",
+        params=_doc_io_method_kwargs_params,
+    )
+    def _read_csv_glob(cls, **kwargs):
+        return cls.io_cls.read_csv_glob(**kwargs)
```

Comment:
You should enable `TestCsvGlob` class with tests for the function on unidist.

Reply:
Marked with `skipif` as Modin on unidist with MPI backend hangs for now.

Reply:
Now unidist on MPI fully functional for Modin so I removed the `skipif` decorator.

unidist 0.2.1 with all necessary fixes will be released shortly.